src/engine/QueryExecutionContext.h

```diff
@@ -48,13 +49,20 @@ class CacheValue {
   [[nodiscard]] size_t size() const {
     return _resultTable ? _resultTable->size() * _resultTable->width() : 0;
   }
+
+  // Calculates the `MemorySize` taken up by an instance of `CacheValue`.
+  struct SizeGetter {
+    ad_utility::MemorySize operator()(const CacheValue& cacheValue) const {
+      return ad_utility::MemorySize::bytes(cacheValue.size() * sizeof(Id));
+    }
+  };
```

Comment:
Move the calculation into the `cacheValue` class  make the `size_ size()` function a `MemorySize consumedMemory()` function or so, and then just call this from the sizeGetter. That gets rid of the strange `size()` function.

Reply:
Sorry, I didn't see that this is a subclass:
Just remove the `size()` function and directly calculate inside the `sizeGetter`.

Reply:
Done, see [046b58f](https://github.com/ad-freiburg/qlever/pull/1111/commits/046b58f8c5645e4ff94ba67429f668d4730966e2).